homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -97,9 +98,20 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
 
     # Create handler
     _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
-    vacuum = Vacuum(host, token)
 
-    mirobo = MiroboVacuum(name, vacuum)
+    try:
+        vacuum = Vacuum(host, token)
+        device_info = vacuum.info()
+        model = device_info.model
+        unique_id = "{}-{}".format(model, device_info.mac_address)
+        _LOGGER.info("%s %s %s detected",
+                     model,
+                     device_info.firmware_version,
+                     device_info.hardware_version)
+    except DeviceException:
+        raise PlatformNotReady
```

Comment:
This will need to allow overriding the model / unique_id somehow, as the device will not respond to `info` if it is not connected to the xiaomi cloud.